app/src/org/commcare/tasks/FormRecordLoaderTask.java

```diff
@@ -260,4 +274,83 @@ public boolean registerPriority(FormRecord record) {
         }
     }
 
+    private static String getTitleFromSession(AndroidSessionWrapper androidSessionWrapper) {
+        //TODO: Most of this mimicks what we need to do in entrydetail activity, remove it from there
+        //and generalize the walking
+
+        // get a copy of the session
+        CommCareSession session = new CommCareSession(androidSessionWrapper.getSession());

```

Comment:
Why do we need to get a copy of the session? Is it ok that its a shallow copy (which I think it is?)